group-coordinator/src/test/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroupTest.java

```diff
@@ -1016,32 +1017,86 @@ public void testMetadataRefreshDeadline() {
         assertEquals(0, group.metadataRefreshDeadline().epoch);
     }
 
-    @ParameterizedTest
-    @ValueSource(booleans = {false, true})
-    public void testValidateOffsetCommit(boolean isTransactional) {
-        ConsumerGroup group = createConsumerGroup("group-foo");
-
-        // Simulate a call from the admin client without member id and member epoch.
-        // This should pass only if the group is empty.
-        group.validateOffsetCommit("", "", -1, isTransactional);
-
-        // The member does not exist.
-        assertThrows(UnknownMemberIdException.class, () ->
-            group.validateOffsetCommit("member-id", null, 0, isTransactional));
-
-        // Create a member.
-        group.updateMember(new ConsumerGroupMember.Builder("member-id").build());
-
-        // A call from the admin client should fail as the group is not empty.
-        assertThrows(UnknownMemberIdException.class, () ->
-            group.validateOffsetCommit("", "", -1, isTransactional));
-
-        // The member epoch is stale.
-        assertThrows(StaleMemberEpochException.class, () ->
-            group.validateOffsetCommit("member-id", "", 10, isTransactional));
+    @Test
+    public void testValidateTransactionalOffsetCommit() {
+        boolean isTransactional = true;
+        for (short v = ApiKeys.OFFSET_COMMIT.oldestVersion(); v <= ApiKeys.OFFSET_COMMIT.latestVersion(); v++) {
+            final short version = v;
+            ConsumerGroup group = createConsumerGroup("group-foo");
+
+            // Simulate a call from the admin client without member id and member epoch.
+            // This should pass only if the group is empty.
+            group.validateOffsetCommit("", "", -1, isTransactional, version);
+
+            // The member does not exist.
+            assertThrows(UnknownMemberIdException.class, () ->
+                group.validateOffsetCommit("member-id", null, 0, isTransactional, version));
+
+            // Create a member.
+            group.updateMember(new ConsumerGroupMember.Builder("member-id").build());
+
+            // A call from the admin client should fail as the group is not empty.
+            assertThrows(UnknownMemberIdException.class, () ->
+                group.validateOffsetCommit("", "", -1, isTransactional, version));
+
+            // The member epoch is stale.
+            assertThrows(StaleMemberEpochException.class, () ->
+                group.validateOffsetCommit("member-id", "", 10, isTransactional, version));
+
+            // This should succeed.
+            group.validateOffsetCommit("member-id", "", 0, isTransactional, version);
+        }
+    }
 
-        // This should succeed.
-        group.validateOffsetCommit("member-id", "", 0, isTransactional);
+    @Test
+    public void testNonTransactionalValidateOffsetCommit() {
```

Comment:
nit: `testValidateOffsetCommit`. If we don't specify transactional, then it is not.